enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,250 @@
+---
+title: cluster-logging-es-rollover-data-design
+authors:
+  - "@jcantril"
+reviewers:
+  - "@bparees"
+  - "@ewolinetz"
+  - "@richm"
+  - "@portante"
+  - "@pavolloffay"
+  - "@sichvoge"
+approvers:
+  - "@ewolinetz"
+  - "@richm"
+creation-date: 2019-11-04
+last-updated: 2019-11-04
+status: implementable
+---
+
+# Cluster Logging Elasticsearch Rollover Data Design
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] ~~Graduation criteria for dev preview, tech preview, GA~~
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This proposal alters the data design for storing logs in ElasticSearch to co-locate logs
+to fewer indices.  It additionally leverages the [Elasticsearch rollover api](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/indices-rollover-index.html) to
```

Comment:
Since v6 of Elasticsearch is rapidly running to end-of-life, can we focus this work on Elasticsearch v7?

Reply:
No.  Only because there is no simple upgrade path from 5 to 7.  We would have to do some trickery to migrate to 6 then 7 which requires us to package both versions and handle it in the operator. More complicated then we have resources to address.... as much as we would prefer to just go straight to 7

Reply:
Why not use the reindex [API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html)?

Reply:
Landed on:
* aliasing the previous indices and depending on DSL to properly secure 
* Allow previous curation to remove old indices over time